lua/xcodebuild/project/manager.lua

```diff
@@ -49,14 +101,28 @@ end
 ---In case of error, it sends an error notification and returns an empty table.
 ---@param action string
 ---@param params string[]|nil
+---@param search_for_project boolean|nil
 ---@return string[]
-local function run(action, params)
+local function run(action, params, search_for_project)
   local allParams = ""
   local project = projectConfig.settings.xcodeproj
   params = params or {}
-  table.insert(params, 1, project)
+  search_for_project = search_for_project or false
+
+  if not search_for_project then
+    table.insert(params, 1, project)
+  end
 
   for _, param in ipairs(params) do
+    if search_for_project then
+      if string.find(param, "/") then
+        local xcodeproj_path = find_xcodeproj_path(param)
+        if xcodeproj_path then
+          allParams = " '" .. xcodeproj_path .. "'" .. allParams
+        end
+      end
+    end
```

Comment:
It looks like it will fail for rename/move where there are two paths provided - it will add xcodeproj path twice.

Maybe I would consider extracting it to another local function like `inject_relative_xcodeproj` and then:

```
if not searchForProject or not inject_relative_xcodeproj(table, params) then
  table.insert(params, 1, project)
end
```

It also fallbacks to the project from settings if the `inject_relative_xcodeproj` fails.